pkg/asset/tls.go

```diff
@@ -111,3 +118,19 @@ func newKubeletKeyAndCert(caCert *x509.Certificate, caPrivKey *rsa.PrivateKey) (
 	}
 	return key, cert, err
 }
+
+func newAPIServerClientKeyAndCert(caCert *x509.Certificate, caPrivKey *rsa.PrivateKey) (*rsa.PrivateKey, *x509.Certificate, error) {
+	key, err := tlsutil.NewPrivateKey()
+	if err != nil {
+		return nil, nil, err
+	}
+	config := tlsutil.CertConfig{
+		CommonName:   "system:api-server",
```

Comment:
Not super familiar with the conventions here - but based on kubelet example (https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/csr/csr.go#L40-L43) seems like the CN should be something like `system:foo:api-sever` (but not sure what 'foo' would become). Then also, would we want this assigned to a particular Organization/Group?